File: src/main/java/leets/weeth/domain/attendance/application/usecase/AttendanceUseCase.java
package leets.weeth.domain.attendance.application.usecase;

import java.util.List;
import leets.weeth.domain.attendance.application.dto.AttendanceDTO;
import leets.weeth.domain.attendance.application.dto.AttendanceDTO.AttendanceInfo;
import leets.weeth.domain.attendance.application.exception.AttendanceCodeMismatchException;

import java.time.LocalDate;

import static leets.weeth.domain.attendance.application.dto.AttendanceDTO.Detail;
import static leets.weeth.domain.attendance.application.dto.AttendanceDTO.Main;

public interface AttendanceUseCase {
    void checkIn(Long userId, Integer code) throws AttendanceCodeMismatchException;

    Main find(Long userId);

    Detail findAllDetailsByCurrentCardinal(Long userId);

    List<AttendanceInfo> findAllAttendanceByMeeting(Long meetingId);

    void close(LocalDate now, Integer cardinal);

    void updateAttendanceStatus(List<AttendanceDTO.UpdateStatus> attendanceUpdates);
}

File: src/main/kotlin/com/weeth/domain/attendance/application/usecase/AttendanceUseCase.kt
package com.weeth.domain.attendance.application.usecase

import com.weeth.domain.attendance.application.dto.request.UpdateAttendanceStatusRequest
import com.weeth.domain.attendance.application.dto.response.AttendanceDetailResponse
import com.weeth.domain.attendance.application.dto.response.AttendanceInfoResponse
import com.weeth.domain.attendance.application.dto.response.AttendanceMainResponse
import java.time.LocalDate

interface AttendanceUseCase {
    fun checkIn(
        userId: Long,
        code: Int,
    )

    fun find(userId: Long): AttendanceMainResponse

    fun findAllDetailsByCurrentCardinal(userId: Long): AttendanceDetailResponse

    fun findAllAttendanceByMeeting(meetingId: Long): List<AttendanceInfoResponse>

    fun close(
        now: LocalDate,
        cardinal: Int,
    )

    fun updateAttendanceStatus(attendanceUpdates: List<UpdateAttendanceStatusRequest>)
}
